src/fabric-custom/web/index.js

```diff
@@ -0,0 +1,58 @@
+//@flow
+import { getIframeId, resizeIframeHeight, getWebfonts, showAdvertLabel } from '../../_shared/js/messages';
+import { write } from '../../_shared/js/dom';
+
+const DapAssetsRoot = `https://s3-eu-west-1.amazonaws.com/adops-assets/dap-fabrics`;
+const DapAssetsFolder = '[%DapAssetsFolder%]';
+
+const DapAssetsPath = `${DapAssetsRoot}/${DapAssetsFolder}`;
+const ThirdPartyTag = '[%ThirdPartyTag%]';
+
+const ifThenPromise = (parameter, action) => {
+  if (parameter === 'true') {
+    return action();
+  } else {
+    return Promise.resolve();
+  }
+}
+
+const fetchTag = (tagUrl: string) => {
+  return fetch(tagUrl)
+    .then(response => response.text())
+}
+
+const replaceAssetLinks = (html: string) => {
+  const re = /url\('\.\/(.*)'\)/g;
+  return html.replace(re, `url('${DapAssetsPath}/$1')`);
+}
+
+const generateTag = () => {
+  if (DapAssetsFolder) {
+    return fetchTag(`${DapAssetsPath}/index.html`)
+      .then(replaceAssetLinks)
+  } else if (ThirdPartyTag) {
+      return fetchTag(`[%ThirdPartyTag%]`)
+  }
+};
+
+const insertTag = (tag) => {
+  const placeholder = document.getElementById('js-fabric-custom');
```

Comment:
You can just do `placeholder.insertAdjacentHTML(“afterbegin”, tag)` instead of using ranges

Reply:
Even if the tag may contain a `<script>` tag?

Reply:
Sorry - I meant to ask: would the script contents be parsed and executed by the browser if I just add it using `insertAdjacentHTML`? As I understand it, I need to use the document fragment / tag pattern to achieve this.

Reply:
once you insert HTML in the DOM tree, using whatever technique (`innerHTML`, `insertAdjacentHTML` or anything really), the browser will go through all the steps to eval the new tree. so yes, if there is a `script` element somewhere, its content will be evaluated by the js engine.

Reply:
Okay, thanks. I'll make that change and test it. Though I'm not sure where I got my original understanding from, [this article](https://ghinda.net/article/script-tags/) represents my original understanding of how this works.

Reply:
oh bummer. You're right. I did not know that :-/

Reply:
I wonder if using `createElement('script')` and then `innerText = <your js>` would work though 🤔  it works when you use the `src` attribute to download a third party script.

Reply:
Ah yeah that might be neater. I'll check and see whether that will work with the kind of tags we're expecting :+1: